main.py

```diff
@@ -1,68 +1,225 @@
-	for el in txt:
-		if el in ESCAPE_CHARS:
-			ans.append("\\")
-		ans.append(el)
-	return ''.join(ans)
-
-def get_mention(user):
-	if user.username == None:
-		return f"[{escape_md(user.first_name)}](tg://user?id={user.id})"
-	else:
-		return "@"+escape_md(user.username)
-
-@message(Filters.status_update.new_chat_members)
-def new_chat_member(update: Update, context: CallbackContext):
-	handles=", ".join(get_mention(user) for user in update.message.new_chat_members)
-	update.message.reply_text(
-f"""{handles},
-いらっしゃいませ\\! \\[Welcome\\!\\]
-Welcome to this chat\\! Please read the rules\\.
-Добро пожаловать в чат\\! Прочти правила, пожалуйста\\.
-このチャットへようこそ！ ルールをお読みください。
-
-[rules](https://t\\.me/dev\\_meme/3667)""",
-parse_mode=ParseMode.MARKDOWN_V2)
-
-print("starting polling")
-updater.start_polling()
-print("online")
-updater.idle()
+#!/usr/bin/env python3
+print("loading libs")
+from os import chdir,path
+from sys import argv
+from datetime import datetime
+from typing import NamedTuple
+from threading import Event, Thread
+import json
+
+from telegram import Update, ParseMode, Chat, User
+from telegram.ext import Updater, CommandHandler, MessageHandler, CallbackContext
+from telegram.ext.filters import Filters
+from telegram.message import Message
+from telegram.utils.helpers import escape_markdown
+import database
+
+
+private_chat_id = -1001218939335
+private_chat_username = 'devs_chat'
+
+print('loading/creating database')
+DB_PATH = 'memebot.db'
+db = database.UserDB(DB_PATH)
+
+def save_db():
+	global db
+	if db.changed:
+		print('saving database')
+		db.changed = False
+		db.dump()
+
+class SaverThread(Thread):
+	'''
+	Class silimar to threading.Timer, but where Timer runs only once, this class runs function multiple times
+
+	To stop the loop, use function set on passed event: event.set()
+	'''
+
+	def __init__(self, event: Event, interval: float):
+		Thread.__init__(self)
+		self.stopped = event
+		self.interval = interval
+
+	def run(self):
+		while not self.stopped.wait(self.interval):
+			save_db()
+
+print('creating saving timer')
+SAVE_INTERVAL = 5 * 60 # 5 minutes
+stop_saving_flag = Event()
+save_timer = SaverThread(stop_saving_flag, SAVE_INTERVAL)
+save_timer.start()
+
+print("reading config")
+CURDIR=path.dirname(argv[0])
+with open(path.join(CURDIR,"./config.json"),"r") as f:
+	CONFIG=json.load(f)
+
+print("initializing commands")
+
+updater=Updater(CONFIG["TOKEN"])
+
+def command(name:str):#some python magic
+	def add_it(func):
+		updater.dispatcher.add_handler(CommandHandler(name, func))
+	return add_it
+
+def message(filters):
+	def add_it(func):
+		updater.dispatcher.add_handler(MessageHandler(filters,func))
+	return add_it
+
+def filter_chat(id, chat):
+	'''
+	id: id of a chat
+	chat: @<chat> without @
+	'''
+	def decorator(function):
+		def wrapper(update: Update, context: CallbackContext):
+			if update.message.chat_id != id:
+				update.message.chat.send_message(f'This feature only works in chat @{chat}')

```

Comment:
maybe add this:

`\n\nIf you want to use this bot outside that group, please contact the developer: @388037461` ← that's my userID

Reply:
Okay :)
Does this id mention work tho?

Reply:
uh, nope, I fucked this one up. It's `[@RiedleroD](tg://user?id=388037461)` (not sure if the extra @ is needed though)